benchmarks/views/index.py

```diff
@@ -140,87 +157,108 @@ def _collect_models(benchmarks, user=None):
-    ScoreDisplay = namedtuple('ScoreDiplay', field_names=[
-        'benchmark', 'benchmark_depth', 'order', 'score_raw', 'score_ceiled', 'error', 'color'])
-    # - prepare "no score" objects for when a model-benchmark score is missing
-    no_score = {}
-    for benchmark in benchmarks:
-        benchmark_identifier = benchmark.identifier
-        benchmark_min, benchmark_max = minmax[benchmark_identifier]
-        no_score[benchmark_identifier] = ScoreDisplay(
-            benchmark=benchmark_identifier, benchmark_depth=benchmark.depth, order=benchmark.overall_order,
-            score_ceiled="", score_raw="", error="",
-            color=representative_color(None, alpha_min=benchmark_min, alpha_max=benchmark_max))
-    # - convert scores DataFrame into rows
-    data = []
-    for model_identifier, group in tqdm(scores.groupby('model'), desc='model rows'):
-        model_scores = {}
-        # fill in computed scores
-        for score_ceiled, score_raw, error, benchmark_identifier in zip(
-                group['score_ceiled'], group['score_raw'], group['error'], group['benchmark']):
-            benchmark_min, benchmark_max = minmax[benchmark_identifier]
-            benchmark = benchmark_lookup[benchmark_identifier]
-            color = representative_color(
-                score_ceiled,
-                colors=colors_redgreen if benchmark.root_parent != 'engineering'
-                else colors_gray,
-                alpha_min=benchmark_min, alpha_max=benchmark_max)
-            score_ceiled = represent(score_ceiled)
-            score_display = ScoreDisplay(benchmark=benchmark_identifier, benchmark_depth=benchmark.depth,
-                                         score_ceiled=score_ceiled, score_raw=score_raw, error=error,
-                                         color=color, order=benchmark.overall_order)
-            model_scores[benchmark_identifier] = score_display
-        # fill in missing scores
-        model_scores = [model_scores[benchmark.identifier] if benchmark.identifier in model_scores
-                        else no_score[benchmark.identifier] for benchmark in benchmarks]
-
-        # put everything together, adding model meta
-        meta = model_meta[model_identifier]
-        model_row = ModelRow(
-            identifier=model_identifier,
-            scores=model_scores, rank=model_ranks[model_ranks['model'] == model_identifier]['rank'].squeeze(),
-            reference_identifier=f"{meta.reference.author} et al., {meta.reference.year}" if meta.reference else None,
-            reference_link=meta.reference.url if meta.reference else None, user=meta.owner, public=meta.public)
-        data.append(model_row)
-    data = list(sorted(data, key=lambda model_row: model_row.rank))
-
-    return data
+    benchmark_lookup = {f'{benchmark.identifier}_v{benchmark.version}': benchmark for benchmark in benchmarks}
+    if scores is not None:
```

Comment:
since this leads to a lot of indentation, can we change this to
```
if scores is None:
    return []
<all the other code here without indent>
```

(also that would make it much easier for me to see the differences in code if there are any)

Reply:
Right, did that.